src/Controllers/DatabaseController.js

```diff
@@ -117,75 +122,6 @@ const validateQuery = query => {
   });
 }
 
-function DatabaseController(adapter, schemaCache) {
```

Comment:
@flovilmart could you fill me on the rationale for the changes in this file? There's a lot that's been shifted around in the diff and I'm not _entirely_ sure what the objective is here.

Reply:
Make it a class, and also make it conform to a particular interface, so it’s easier to know what’s consumed and how in the future (like the indexing) this also rationalizes to a certain extent also the interfaces for every calls to the adapters.

Reply:
Ah, now I see! Thanks for clarifying.